hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -736,23 +744,48 @@ public void notifyTermIndexUpdated(long term, long index) {
     return f;
   }
 
+  // Removes the stateMachine data from cache once both followers catch up
+  // to the particular index.
+  private void removeStateMachineDataIfNeeded(long index) {
+    if (waitOnBothFollowers) {
```

Comment:
We will need to handle the case when waitOnBothFollowers is false.

Reply:
waitOnBothFollowers is false by default and it should revert to the current behaviour if the condition is false. Will fix this.